### PR TITLE
Bugfixes for install_ubuntu.sh

### PR DESCRIPTION
**Pull request by ph4ge**
The minor changes proposed for the install_ubuntu.sh script concern:
- handling _curl_ dependency for Rust before trying to install it + making the shell environment aware of cargo install
- the _make install_ for capstone-4.0.2 requires elevated privileges in order to write to _/usr/lib_

### CODE REVIEW

**Comment by AndrewFasano** [comment]
Hey, these changes look great! I'll let the CI run and once it (presumably) passes, will happily merge. Thanks for the contribution!

**Comment by ph4ge** [comment]
Hi Andrew,

I also noticed that if I didn't perform a complete  apt update && upgrade before the running install_ubuntu.sh
I have a dependency issue, i should have the package information somewhere:

The following packages have unmet dependencies:
 libepoxy-dev : Depends: libegl1-mesa-dev but it is not going to be installed
                Depends: libgl1-mesa-dev but it is not going to be installed
E: Unable to correct problems, you have held broken packages.

To recap, I observe the aforementioned message(libepoxy-dev) if i try launching the install_ubuntu.sh script without having performed a full update && upgrade before.

But I suppose everybody does this. 

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by AndrewFasano** [comment]
Hmm. The `apt update` happens automatically in the script which I think should normally set the subsequent apt commands up to succeed. Perhaps something had gotten messed up on your system prior to running the script? Or were you using a brand new system (or vm or container)? I worry that running `apt upgrade` automatically might surprise some users who didn't want other packages on their system upgraded.

**Comment by ph4ge** [comment]
I used a freshly installed system in virtualbox.
I share your concern about the upgrade stuff, that's why I haven't touched it.
Anyway, the "modifications" don't touch/modify "core" stuff.

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by AndrewFasano** [comment]
Hm, okay. I'll have to think about that then, not sure what would be the best way to fix that case then. For what it's worth, our [CI job that tests install_ubuntu.sh](https://github.com/panda-re/panda/blob/dev/.github/workflows/parallel_tests.yml#L19-L24) just starts from a clean ubuntu 20.04 container and runs the install_ubuntu script - not sure why that would behave differently from your system.

But we don't have to figure out exactly why your system needed the `apt upgrade` to merge this PR so I'll do that now :) Thanks again for the contribution
